src/bin/cql-stress-cassandra-stress/settings/param/parser.rs

```diff
@@ -0,0 +1,189 @@
+        Self {
+            command_name: command_name.to_owned(),
+            params: Vec::new(),
+            groups: Vec::new(),
+        }
+    }
+
+    /// Registers the simple parameter provided by the user and returns the handle.
+    /// `value_pattern` has to be a regular expression, otherwise we panic.
+    pub fn simple_param(
+        &mut self,
+        prefix: &'static str,
+        value_pattern: &'static str,
+        default: Option<&'static str>,
+        desc: &'static str,
+        required: bool,
+    ) -> SimpleParamHandle {
+        let param = Rc::new(RefCell::new(SimpleParam::new(
+            prefix,
+            value_pattern,
+            default,
+            desc,
+            required,
+        )));
+
+        self.params.push(Rc::clone(&param) as ParamCell);
+        SimpleParamHandle::new(param)
+    }
+
+    /// Creates a new group of the parameters.
+    pub fn group(&mut self, params: Vec<&dyn ParamHandle>) {
+        self.groups.push(ParamsGroup::new(
+            params.into_iter().map(|handle| handle.cell()).collect(),
+        ))
+    }
+
+    fn parse_groups(&mut self, args: Vec<&str>) -> Result<()> {
+        for arg in args {
+            let mut consumed = false;
+            for group in self.groups.iter_mut() {
+                match group.parse(arg) {
+                    ParamParseResult::Success => {
+                        consumed = true;
+                        break;
+                    }
+                    ParamParseResult::Error(e) => {
+                        return Err(e);
+                    }
+                    _ => (),
+                }
+            }
+            anyhow::ensure!(consumed, "Invalid parameter {}", arg);
+        }
+
+        Ok(())
+    }
+
+    // Consume the parser during parsing.
+    pub fn parse(mut self, args: Vec<&str>) -> Result<()> {
+        if self.groups.is_empty() {
+            // User didn't specify any groups. Treat all parameters as a single group.
+            self.groups.push(ParamsGroup::new(self.params.clone()));
+        }
+
+        let args_size = args.len();
+        self.parse_groups(args)?;
+        if self.groups.iter().any(|g| g.satisfied(args_size)) {
+            return Ok(());
+        }
+
+        Err(anyhow::anyhow!(
+            "Invalid {} parameters provided, see `help {}` for valid parameters",
+            self.command_name.to_uppercase(),
+            self.command_name
+        ))
+    }
+
+    pub fn print_help(&self) {
+        let groups_size = self.groups.len();
+        println!();
+        for (i, group) in self.groups.iter().enumerate() {
+            print!("Usage: {} ", self.command_name);
+            group.print_help();
+            if i < groups_size - 1 {
+                print!("\n OR");
+            }
+            println!();
+        }
+        println!();
+
+        for param in self.params.iter() {
+            print!("  ");
+            param.borrow().print_desc();
+        }
+    }
+}
```

Comment:
A test here or in `simple_param.rs` would be appreciated - it would also show the relationship between param and handle.